### PR TITLE
Adding reviewers.py to help tag reviewers in commit message

### PR DESCRIPTION
**Pull request by mumrah**
This is a simple python script which looks at the commit log and lets you easily find names + emails for the "Reviewers" line.

You can search by first name or email. Here's a sample session:

```
$ ./reviewers.py 
Utility to help generate 'Reviewers' string for Pull Requests. Use Ctrl+D or Ctrl+C to exit

Name or email (case insensitive): Dav <-- Just need a few letters

Possible matches (in order of most recent):
[1] David Jacot djacot@confluent.io (141)  <-- Shows a count of occurrences in the commit log
[2] David Arthur mumrah@gmail.com (102)
[3] David Arthur david.arthur@confluent.io (5)
[4] David Jacot david.jacot@gmail.com (8)

Make a selection: 2
Reviewers so far: [('David Arthur', 'mumrah@gmail.com', 102)]

Name or email (case insensitive): chia

Possible matches (in order of most recent):
[1] Chia-Ping Tsai chia7712@gmail.com (358)
[2] Chia-Ping Tsai chia7712@apache.org (3)

Make a selection: 1
Reviewers so far: [('David Arthur', 'mumrah@gmail.com', 102), ('Chia-Ping Tsai', 'chia7712@gmail.com', 358)]

Name or email (case insensitive): ism

Possible matches (in order of most recent):
[1] Ismael Juma ismael@juma.me.uk (1514)
[2] Ismael Juma ijuma@apache.org (3)
[3] Ismael Juma mlists@juma.me.uk (4)
[4] Ismael Juma ismael@confluent.io (19)
[5] Ismael Juma github@juma.me.uk (7)

Make a selection: 1
Reviewers so far: [('David Arthur', 'mumrah@gmail.com', 102), ('Chia-Ping Tsai', 'chia7712@gmail.com', 358), ('Ismael Juma', 'ismael@juma.me.uk', 1514)]

Name or email (case insensitive): ^C

Reviewers: David Arthur <mumrah@gmail.com>, Chia-Ping Tsai <chia7712@gmail.com>, Ismael Juma <ismael@juma.me.uk>
```

ctrl+d or ctrl+c will exit the program and print out the "Reviewers" line based on the selection.

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks, this is helpful. Any chance we can add a map that deduplicates emails from committers? In this PR, we could add the map and a couple of cases and then we can iterate as needed.

**Comment by mumrah** [comment]
@ijuma I've been using this script for a while now with good results. Can we merge this and defer improvements to future PRs?

File: reviewers.py
Comment (jsancio) [inline reviewers.py:24]: How about this since you are `lstrip("Reviewers: ")` later in the code?
```python
    stream = os.popen('git log --grep "Reviewers: " --format=%b | grep "Reviewers: "')
```

File: reviewers.py
Comment (jsancio) [inline reviewers.py:58]: Can we print it using the standard `name <email>` address format? E.g.
```python
            print(f"[{i+1}] {candidate[0]} <{candidate[1]}> ({candidate[2]})")
```

File: reviewers.py
Comment (jsancio) [inline reviewers.py:56]: Should we mention that the number in the parenthesis is the number of occurrences? How about something like:
```python
        print("\nMatches order by most recent (the number in the parenthesis is the number of occurrences):")
```
or we can change the formatting to something like:
```
Name <email> (reviews=123)
```

File: reviewers.py
Comment (jsancio) [inline reviewers.py:22]: Both Ctrl+C and Ctrl+D print the selected reviewers? If not, we should mention that.

Either way, I think we should mention that they need to Ctrl+D to print the selected reviewers.

**Review by jsancio — COMMENTED** [review]
Thanks for the tool @mumrah. I would love to use this tool.

File: reviewers.py
Comment (showuon) [inline reviewers.py:13]: nit: Since we're just checking the prefix of the name (or email), not sure if we should mention that here.

File: reviewers.py
Comment (showuon) [reply reviewers.py:56]: Should we mention we only list top 10 candidates?

**Review by showuon — APPROVED** [review]
Thanks for the great tool! Left some minor comments. I'm good whether you want to address them here or in the follow-up PR.

File: reviewers.py
Comment (mumrah) [reply reviewers.py:56]: @showuon the zip + range(10) has the effect of only showing the top 10

**Comment by mumrah** [comment]
@jsancio @showuon thanks for taking a look! Here's what a session looks like after the suggested changes:

```
Utility to help generate 'Reviewers' string for Pull Requests.
Use Ctrl+D or Ctrl+C to print the selected reviewers and exit.

Name or email (case insensitive, partial matches allowed): david

Matches in order by most recent. Format is: [Selection] Name <Email> (Occurrences):
[1] David Jacot <djacot@confluent.io> (315)
[2] David Arthur <mumrah@gmail.com> (177)
[3] David Jacot <david.jacot@gmail.com> (12)
[4] David Mao <dmao@confluent.io> (4)
[5] David Arthur <david.arthur@confluent.io> (5)

Make a selection: 2
Reviewers so far: [('David Arthur', 'mumrah@gmail.com', 177)]

Name or email (case insensitive, partial matches allowed): josé

Matches in order by most recent. Format is: [Selection] Name <Email> (Occurrences):
[1] José Armando García Sancio <jsancio@apache.org> (19)
[2] José Armando García Sancio <jsancio@gmail.com> (38)
[3] José Armando García Sancio <jsancio@users.noreply.github.com> (72)

Make a selection: 1
Reviewers so far: [('David Arthur', 'mumrah@gmail.com', 177), ('José Armando García Sancio', 'jsancio@apache.org', 19)]

Name or email (case insensitive, partial matches allowed): luke

Matches in order by most recent. Format is: [Selection] Name <Email> (Occurrences):
[1] Luke Chen <showuon@gmail.com> (310)
[2] Luke Chen <showuon@confluent.io> (4)

Make a selection: 1
Reviewers so far: [('David Arthur', 'mumrah@gmail.com', 177), ('José Armando García Sancio', 'jsancio@apache.org', 19), ('Luke Chen', 'showuon@gmail.com', 310)]

Name or email (case insensitive, partial matches allowed): ^D

Reviewers: David Arthur <mumrah@gmail.com>, José Armando García Sancio <jsancio@apache.org>, Luke Chen <showuon@gmail.com>
```

**Comment by jolshan** [comment]
@mumrah Can we please check builds before merging? The build caught the license issue.

**Pull request merged**
